internal/util/indexcgowrapper/dataset.go

```diff
@@ -0,0 +1,103 @@
+package indexcgowrapper
+
+import (
+	"github.com/milvus-io/milvus/internal/proto/schemapb"
+	"github.com/milvus-io/milvus/internal/storage"
+)
+
+const (
+	keyRawArr = "key_raw_arr"
```

Comment:
what is the core reason we cant to make Data a map here? should it be simply a list or slice?

Reply:
no other core reason, I just borrowed the design of our c++ interface. interface `Build(Dataset)` correspond to c++ `Build(Dataset)`. In c++ implementations, `Dataset` was defined as a map.